ci: Add GitHub workflow for linting and golangci-lint

OdedViner opened the pull request:
Example:
https://github.com/OdedViner/ramenctl/actions/runs/13816226385/job/38650016161

nirs commented on hack/pre-commit.sh:
I don't want these kind bash scripts in this project. Lets focus only on golangci-lint for now and remove everything else.

nirs commented on .github/workflows/ramen.yaml:
```diff
@@ -0,0 +1,82 @@
+  run:
+    shell: bash
+jobs:
+  lint:
```
Lets remove this for now, I want to add now only golangci-lint. We can consider yamllint later. I don't think we will add markdown linter now, and if we add it will be using the default.

nirs commented on .github/workflows/ramen.yaml:
```diff
@@ -0,0 +1,82 @@
+      - name: Run linters
+        run: ./hack/pre-commit.sh
+
+      - name: Check tidiness
```
We don't have generated files so the usefulness of this questionable. Please do not copy entire jobs from ramen, this way we create a big mess, adding lot of stuff we don't really need. I want to add only stuff we care about and in small steps when we completely understand every piece of configuration and code.

nirs commented on .github/workflows/ramen.yaml:
```diff
@@ -0,0 +1,82 @@
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get install -y python3-pip ruby
+          sudo gem install mdl
+          sudo pip3 install yamllint
```
I think this complexity is not needed on github. Lets remove it for now and add only the parts we need now.

nirs commented on .github/workflows/ramen.yaml:
```diff
@@ -0,0 +1,82 @@
+      - name: GolangCI Lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.62.0
```
Use latest version here.

nirs commented on .github/workflows/ramen.yaml:
```diff
@@ -0,0 +1,82 @@
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.62.0
+          working-directory: .
```
Do we need this?

nirs commented on hack/mdl-style.rb:
Remove fro this PR.

nirs commented on hack/yamlconfig.yaml:
Remove from this PR

nirs commented on .github/workflows/ramen.yaml:
This file came from ramen when the name makes sense. Let's rename to "test.yaml"

nirs reviewed (commented):
Starting with golangci-lint using the defaults is good. We can add the `.golangci-lint.yaml` later.

The rest of the checks are not needed now. Lets focus only on having a golangci-lint job for now.

OdedViner commented:
@nirs I have addressed your comments in the PR and made the necessary changes. Please review the updates when you have time. Feel free to let me know if there's anything else I should adjust.  https://github.com/OdedViner/ramenctl/actions/runs/13894547777/job/38872241003

nirs commented:
@OdedViner PR was replaced by #113
